python/dolfinx/io/gmshio.py

```diff
@@ -60,7 +60,7 @@ def ufl_mesh(gmsh_cell: int, gdim: int, dtype=npt.DTypeLike) -> ufl.Mesh:
     cell = ufl.Cell(shape, geometric_dimension=gdim)
     element = basix.ufl.element(basix.ElementFamily.P, cell.cellname(), degree,
                                 basix.LagrangeVariant.equispaced, shape=(gdim,),
-                                gdim=gdim, dtype=dtype)
+                                gdim=gdim, dtype=dtype)  # type: ignore[arg-type]
```

Comment:
`basix.ufl.element` is decorated with `_functools.lru_cache()`, and there doesn't seem to be a way to make that decorator behave nicely with `mypy` for python < 3.11, see https://github.com/python/mypy/issues/5107, hence the ignore.